jdaviz/app.py

```diff
@@ -311,16 +311,31 @@ def _link_new_data(self):
 
         # Can't link if there's no world_component_ids
         wc_new = self.data_collection[-1].world_component_ids
-        if wc_new == []:
-            return
+        pc_new = self.data_collection[-1].pixel_component_ids
 
         # Link to the first dataset with compatible coordinates
         for i in range(new_len - 1):
             wc_old = self.data_collection[i].world_component_ids
+            pc_old = self.data_collection[i].pixel_component_ids
+
             if wc_old == []:
                 continue
             else:
-                self.data_collection.add_link(LinkSame(wc_old[0], wc_new[0]))
+                # Link cubes to each other
+                if len(pc_old) > 2 and len(pc_new) > 2:
```

Comment:
Do we actually still need this now that you are linking in the plugin?

Reply:
Yes, in main cubes also cannot utilize contour over-plotting because they are not linked by pixels in the x and y coordinates.

Reply:
Now I am super confused... Haha. What is the one in the plugin for then?

Reply:
Okay, I think I get it... this one here is for cube on cube... ?

Reply:
Correct!